core/templates/pages/contributor-dashboard-page/contributor-dashboard-page.component.ts

```diff
@@ -16,219 +16,214 @@
-            ctrl.username = '';
-          }
-          // TODO(#8521): Remove the use of $rootScope.$apply()
-          // once the controller is migrated to angular.
-          $rootScope.$applyAsync();
-        });
-
-        ctrl.topicName = DEFAULT_OPPORTUNITY_TOPIC_NAME;
-        TranslationTopicService.setActiveTopicName(ctrl.topicName);
-
-        ContributionOpportunitiesService.getTranslatableTopicNamesAsync()
-          .then(function(topicNames) {
-            if (topicNames.indexOf(prevSelectedTopicName) !== -1) {
-              ctrl.topicName = prevSelectedTopicName;
-              TranslationTopicService.setActiveTopicName(ctrl.topicName);
-            }
-            $rootScope.$applyAsync();
-          });
-
-        ctrl.activeTabName = 'myContributionTab';
-        ctrl.tabsDetails = CONTRIBUTOR_DASHBOARD_TABS_DETAILS;
-        ctrl.OPPIA_AVATAR_LINK_URL = OPPIA_AVATAR_LINK_URL;
-        ctrl.OPPIA_AVATAR_IMAGE_URL = (
-          UrlInterpolationService.getStaticImageUrl(
-            '/avatar/oppia_avatar_100px.svg'));
-      };
+import { Component, OnInit } from '@angular/core';
+import { SafeUrl } from '@angular/platform-browser';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { LanguageUtilService } from 'domain/utilities/language-util.service';
+import AppConstants from 'assets/constants';
+import { ContributorDashboardConstants, ContributorDashboardTabsDetails } from 'pages/contributor-dashboard-page/contributor-dashboard-page.constants';
+import { ContributionAndReviewService } from './services/contribution-and-review.service';
+import { ContributionOpportunitiesService } from './services/contribution-opportunities.service';
+import { FocusManagerService } from 'services/stateful/focus-manager.service';
+import { LocalStorageService } from 'services/local-storage.service';
+import { SvgSanitizerService } from 'services/svg-sanitizer.service';
+import { TranslationLanguageService } from 'pages/exploration-editor-page/translation-tab/services/translation-language.service';
+import { TranslationTopicService } from 'pages/exploration-editor-page/translation-tab/services/translation-topic.service';
+import { UserService } from 'services/user.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+@Component({
+  selector: 'contributor-dashboard-page',
+  templateUrl: './contributor-dashboard-page.component.html'
+})
+export class ContributorDashboardPageComponent
+  implements OnInit {
+  defaultHeaderVisible: boolean;
+  profilePictureDataUrl: SafeUrl | string;
+  username: string;
+  userInfoIsLoading: boolean;
+  userIsLoggedIn: boolean;
+  userIsReviewer: boolean;
+  userCanReviewTranslationSuggestionsInLanguages: unknown[];
+  userCanReviewVoiceoverSuggestionsInLanguages: unknown[];
+  userCanReviewQuestions: boolean;
+  OPPIA_AVATAR_LINK_URL: string;
```

Comment:
Can we move the declaration of this constant and `OPPIA_AVATAR_IMAGE_URL` to the top (line 42) and initialize them from `AppConstants` here instead of in `ngOnInit`?

Reply:
done